homeassistant/components/prometheus/__init__.py

```diff
@@ -235,10 +252,14 @@ def handle_entity_registry_updated(self, event):
         if metrics_entity_id:
             self._remove_labelsets(metrics_entity_id)
 
-    def _remove_labelsets(self, entity_id, friendly_name=None):
+    def _remove_labelsets(
+        self, entity_id: str, friendly_name: str | None = None
+    ) -> None:
         """Remove labelsets matching the given entity id from all metrics."""
         for _, metric in self._metrics.items():
-            for sample in metric.collect()[0].samples:
+            for sample in cast(list[prometheus_client.Metric], metric.collect())[
```

Comment:
`cast` is necessary due to an upstream issue. `metric.collect()` is annotated to return an `Iterable` even though it actually returns a list with exactly one value.

`__getitem__` lookups require at least a `Sequence` and thus don't work for `Iterable`.

https://github.com/prometheus/client_python/blob/v0.17.1/prometheus_client/metrics.py#L99-L103